packages/server/dataloader/customLoaderMakers.ts

```diff
@@ -843,58 +843,62 @@ export const meetingCount = (parent: RootDataLoader, dependsOn: RegisterDependsO
-//     FeatureFlag[],
-//     string
-//   >(
-//     async (keys) => {
-//       const pg = getKysely()
-
-//       const userIds = keys.filter((k) => k.scope === 'User').map((k) => k.ownerId)
-//       const teamIds = keys.filter((k) => k.scope === 'Team').map((k) => k.ownerId)
-//       const orgIds = keys.filter((k) => k.scope === 'Organization').map((k) => k.ownerId)
-
-//       let query = pg
-//         .selectFrom('FeatureFlagOwner')
-//         .innerJoin('FeatureFlag', 'FeatureFlag.id', 'FeatureFlagOwner.featureFlagId')
-//         .selectAll('FeatureFlag')
-
-//       if (userIds.length > 0) {
-//         query = query.where('FeatureFlagOwner.userId', 'in', userIds)
-//       }
-
-//       if (teamIds.length > 0) {
-//         query = query.where('FeatureFlagOwner.teamId', 'in', teamIds)
-//       }
-
-//       if (orgIds.length > 0) {
-//         query = query.where('FeatureFlagOwner.orgId', 'in', orgIds)
-//       }
-
-//       const featureFlags = await query.execute()
-
-//       const featureFlagsByOwner = new Map<string, FeatureFlag[]>()
-//       for (const key of keys) {
-//         featureFlagsByOwner.set(
-//           `${key.ownerId}:${key.scope}`,
-//           featureFlags.filter((flag) => {
-//             return (
-//               (key.scope === 'User' && flag.userId === key.ownerId) ||
-//               (key.scope === 'Team' && flag.teamId === key.ownerId) ||
-//               (key.scope === 'Organization' && flag.orgId === key.ownerId)
-//             )
-//           })
-//         )
-//       }
-
-//       return keys.map((key) => featureFlagsByOwner.get(`${key.ownerId}:${key.scope}`) || [])
-//     },
-//     {
-//       ...parent.dataLoaderOptions,
-//       cacheKeyFn: (key) => `${key.ownerId}:${key.scope}`
-//     }
-//   )
-// }
+export const featureFlagByOwnerId = (parent: RootDataLoader) => {
+  return new DataLoader<{ownerId: string; featureName: string}, boolean, string>(
+    async (keys) => {
+      const pg = getKysely()
+
+      const featureNames = [...new Set(keys.map(({featureName}) => featureName))]
+      const ownerIds = [...new Set(keys.map(({ownerId}) => ownerId))]
+
+      if (!__PRODUCTION__) {
+        const existingFeatureNames = await pg
+          .selectFrom('FeatureFlag')
+          .select('featureName')
+          .where('featureName', 'in', featureNames)
+          .execute()
+
+        const existingFeatureNameSet = new Set(existingFeatureNames.map((row) => row.featureName))
+
+        const missingFeatureNames = featureNames.filter((name) => !existingFeatureNameSet.has(name))
+        if (missingFeatureNames.length > 0) {
+          throw new Error(`Feature flag name(s) not found: ${missingFeatureNames.join(', ')}`)
+        }
+      }
+
+      const results = await pg
+        .selectFrom('FeatureFlag')
+        .innerJoin('FeatureFlagOwner', 'FeatureFlag.id', 'FeatureFlagOwner.featureFlagId')
+        .where((eb) =>
+          eb.and([
```

Comment:
+1 you could simplify this a little because FeatureFlag tells you what column to look in so you don't have to look in all 3 columns.

You also don't have to destructure the `(featureName, ownerId)` tuple. You could do something like:

```
WHERE 
  ("featureName", CASE "FeatureFlag"."scope"
    WHEN 'User' THEN "FeatureFlagOwner"."userId"
    WHEN 'Team' THEN "FeatureFlagOwner"."teamId"
    WHEN 'Organization' THEN "FeatureFlagOwner"."orgId"
  END) in :keys;
``` 

This will execute as something like `WHERE ("featureName", "userId") in (("insights", "user123"))`. That way you won't overfetch (albeit the overfetch is pretty minimal in this case)